[Mukund Raja] iP

mukundrs opened the pull request:
# Wednesday Dude
> “It's wednesday my dudes” – Wednesday dude

**Wednesday Dude** is an application that allows you to manage your tasks more _efficiently_!!! 😄

* Text-based
* Quick and easy to use
* ~Easy~ Very easy to learn!
* USE TODAY!!! :joy: :joy: :joy:

## Features

- [x] Add tasks such as todos, deadlines or events
- [X] Mark them as done
- [X] Find tasks using specific key words
- [X] Delete tasks
- [X] List out all your tasks

## Setup

Prerequisites: JDK 11, update Intellij to the most recent version.

1. Click [here](https://github.com/mukundrs/ip/releases/tag/A-BetterGui) and download the jar file.
2. Run java -jar ip.jar in command prompt. Yoou should see this output: 
  
[Image of Ui](https://github.com/mukundrs/ip/blob/master/docs/Ui.png)


If you Java programmer, you can use it to practice Java too. Here's the `Launcher` class:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```
Adopted and modified from [Project Duke](https://github.com/nus-cs2103-AY2122S1/ip)

Commit:
Add Gradle support

mukundrs committed:
Level-1

mukundrs committed:
Level-2

mukundrs committed:
Level-3

mukundrs committed:
Level-3

mukundrs committed:
Level-4

mukundrs committed:
Level-4

mukundrs committed:
Level-4

mukundrs committed:
A-TextUiTesting

mukundrs committed:
Level-5

mukundrs committed:
Level-6

mukundrs committed:
Added save to hard disk functionality

mukundrs committed:
Added bug fix

mukundrs committed:
Added minor bug fix

mukundrs committed:
Added minor bug fixes

mukundrs committed:
Added dateTime functionality

mukundrs committed:
Merge tag 'branch-Level-8'

mukundrs committed:
Refactored out code to extract out closely related classes

mukundrs committed:
Removed unnecessary comments

mukundrs committed:
Added JUnit tests

mukundrs committed:
Added Jar

mukundrs committed:
Added JavaDocs

mukundrs committed:
Made changes adhering to coding standard

mukundrs committed:
Added find functionality

mukundrs committed:
Merge tag 'A-CodingStandard'

mukundrs committed:
Merge tag 'branch-Level-9'

boonhaii commented on src/main/java/duke/DukeStorage.java:
```diff
@@ -0,0 +1,127 @@
+                    String[] parseTask = str.split(" \\| ");
+
+                    String taskType = parseTask[0];
+                    boolean isDone = parseTask[1].equals("1") ? true : false;
```
Perhaps you could exclude the need for a ternary operation since equals() method already returns a boolean.
```suggestion
                    boolean isDone = parseTask[1].equals("1");
```

boonhaii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,105 @@
+package duke;
+
+import duke.commands.*;
```
Maybe you could list the imports explicitly since the Java Coding Standards recommend not doing wildcard imports due to possible name conflicts.

michael-lee-sk replied:
Agree with @boonhaii that you should list the imports explicitly

>  Imported classes should always be listed explicitly. [Source](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#package-and-import-statements)

boonhaii commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+     * @return Deleted task
+     */
+    public Task delete(int index) {
+        Task deletedTask = this.taskList.get(index - 1);
+        this.taskList.remove(index - 1);
```
Perhaps you could combine these two lines into one? Since ArrayList.remove() method returns the element at the index which you removed.
```suggestion
        Task deletedTask = this.taskList.remove(index - 1);
```

boonhaii commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+    @Override
+    public String toString() {
+        String tasks = "";
+        for (int i = 0; i < counter; i++) {
```
Perhaps we can use this.counter rather than just counter to indicate that counter is a member and not a standalone variable?
```suggestion
        for (int i = 0; i < this.counter; i++) {
```

boonhaii commented on src/main/java/duke/commands/DoneCommand.java:
```diff
@@ -0,0 +1,54 @@
+     * @param taskList The list of tasks that is associated with the instance of Duke
+     * @param ui The UI that is associated with the instance of Duke
+     * @param storage The storage that is associated with the instance of Duke
+     * @throws DukeException
```
I think you may have accidentally missed the description for the throw tag in this method's Javadoc

boonhaii reviewed (commented):
Looks good! Just a few nits to fix!
Overall your code was easy to read. There were a few minor coding standard violations here and there. I did feel that in certain files such as Duke.java, there were lots of empty lines which were placed to help improve readability, but I personally felt it unnecessarily elongated the code, which made it harder to read. Maybe you could also consider removing certain empty lines if you feel they are not essential?

michael-lee-sk commented on src/main/java/duke/DukeStorage.java:
```diff
@@ -0,0 +1,127 @@
+public class DukeStorage {
+
+    /** The destination of the stored data */
+    public String path;
```
Maybe you could set String path to private, and create a getter in this class. Would help with your encapsulation
```suggestion
    private String path;
```

michael-lee-sk commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import duke.tasks.Task;
+
```
remove the empty space?

michael-lee-sk commented on src/test/java/CommandTest.java:
```diff
@@ -0,0 +1,53 @@
+import duke.Ui;
+import duke.TaskList;
+import duke.DukeStorage;
+
```
remove the empty space here?

michael-lee-sk reviewed (commented):
LGTM, just some small portions to relook at. Thought your code was pretty clean and neat, making it pretty easy to read. Encapsulation and OOP principals were rather good. On top of @boonhaii, I think you did a pretty neat job 👍

mukundrs committed:
Update README.md

mukundrs committed:
Update README.md

mukundrs committed:
Update README.md

mukundrs committed:
Update README.md

mukundrs committed:
Merge branch 'add-gradle-support'

mukundrs committed:
Merge branch 'add-gradle-support' into A-Gradle

mukundrs committed:
Merge branch 'A-Gradle'

mukundrs committed:
Added checkstyle and made some checkstyle edits

mukundrs committed:
Added GUI

mukundrs committed:
Add assertions

Add assertions to document important assumptions that hold at various points of code

Added for extra error handling

mukundrs committed:
Merge pull request #2 from mukundrs/A-Assertions

Add assertions

mukundrs committed:
Improve Code Quality

Updated code quality to improve readability

Improved JavaDocs adhering to JavaDocs standards

mukundrs committed:
Merge pull request #3 from mukundrs/A-CodeQuality

Improve Code Quality

mukundrs committed:
Add extension C-Help

Add C-Help functionality to:
-Guide user on using Duke
-Inform user of Duke syntax

mukundrs committed:
Merge pull request #4 from mukundrs/C-Help

Add extension C-Help

mukundrs committed:
Update GUI and minor checkStyle fix

Update GUI to make it look more appealing and fix checkStyle
to improve readability

mukundrs committed:
Add ui screenshots

mukundrs committed:
Update README.md

mukundrs committed:
Update README.md

mukundrs committed:
Add ui screenshots

mukundrs committed:
Merge branch 'master' of https://github.com/mukundrs/ip

mukundrs committed:
Update README.md

mukundrs committed:
Update Build.Gradle

mukundrs committed:
Update README.md

mukundrs committed:
Update README.md

mukundrs committed:
Update README.md

mukundrs committed:
Update README.md
